src/components/Link/index.tsx

```diff
@@ -38,35 +38,39 @@ export const Link = forwardRef<HTMLAnchorElement, LinkProps<unknown>>((({ lng, t
 
 export type BTCExplorerLinkProps<S> =
   | (Omit<RouterLinkProps<S>, 'to'> & {
-      id: string
+      id?: string
+      address?: string
       path: string
       lng: SupportedLng
       anchor?: string
     })
   | {
-      id: string
+      id?: string
+      address?: string
       path: string
       lng?: SupportedLng
       anchor?: string
     }
 
 export const BTCExplorerLink = forwardRef<HTMLAnchorElement, BTCExplorerLinkProps<unknown>>(((
-  { id, lng, path, anchor, ...props },
+  { id, address, lng, path, anchor, ...props },
   ref,
 ) => {
   const { locale } = useParams<{ locale?: string }>()
   const { data: identity } = useQuery({
-    queryKey: ['btc-testnet-identity', id],
-    queryFn: () => (id ? getBtcChainIdentify(id) : null),
-    enabled: !IS_MAINNET && !!id && path === '/tx',
+    queryKey: ['btc-testnet-identity', id, address],
+    queryFn: () => getBtcChainIdentify(id!),
+    enabled: !IS_MAINNET && !!id,
   })
 
   return (
     <Link
       lng={lng ?? (locale as 'en' | 'zh')}
       ref={ref}
       {...props}
-      to={`${config.BITCOIN_EXPLORER}${IS_MAINNET ? '' : `/${identity}`}${path}/${id}${anchor ? `#${anchor}` : ''}`}
+      to={`${config.BITCOIN_EXPLORER}${IS_MAINNET ? '' : `/${identity ?? 'testnet'}`}${path}/${address ?? id}${
```

Comment:
It's better to assign the fallback default value where `identity` is set so the default value won't be missed elsewhere.

Reply:
> It's better to assign the fallback default value where `identity` is set so the default value won't be missed elsewhere.

This is a basic component, it won't to pass the identity to any other components, so I think it is not necessary, here the default value obvious is the "testnet".

Reply:
We have to remember `?? 'testnet'` when `identity` is used elsewhere in this component.
If `identity` was set with a default `testnet` when it's defined, for example, at https://github.com/Magickbase/ckb-explorer-frontend/pull/412/files/603e800939a0482a24b285853666d9d22bd0454f#diff-f4edd6698ac8ef92e05ed39a633d29fd9c35ac226d25bebf00eb6d95fd8f1214L58, we don't have to remember it.

Reply:
> > We have to remember `?? 'testnet'` when `identity` is used elsewhere in this component. If `identity` was set with a default `testnet` when it's defined, for example, at [Magickbase/ckb-explorer-frontend/pull/412/files/603e800939a0482a24b285853666d9d22bd0454f#diff-f4edd6698ac8ef92e05ed39a633d29fd9c35ac226d25bebf00eb6d95fd8f1214L58](https://github.com/Magickbase/ckb-explorer-frontend/pull/412/files/603e800939a0482a24b285853666d9d22bd0454f#diff-f4edd6698ac8ef92e05ed39a633d29fd9c35ac226d25bebf00eb6d95fd8f1214L58), we don't have to remember this detail.
> 
> It might be null, and why to remember this, I didn't understand.

When other programmers add codes to this component, they should pay attention to use `identity` with `?? 'testnet'` to keep all references consistent.

Say we are going to add a mark when it's on testnet, `<badge>{identity ?? 'testnet'}</badge>` should be added, in the case that the coder has noticed it.

If the coder is reckless, he/she may add `<badge>{identity}</badge>` directly, which is different from `identity ?? 'testnet'`

If the `identity` is defined with a default value `'testnet'`, `<badge>{identity}</badge>` can be used directly.